shaman/tests/controllers/test_search.py

```diff
@@ -63,3 +70,57 @@ def test_filter_no_matching_distro(self, session):
         session.app.post_json('/api/repos/ceph/', params=base_repo_data())
         result = session.app.get('/api/search/', params={'distros': 'centos/7'})
         assert result.json == []
+
+
+class TestLatestSha1(object):
+
+    def test_single_repo_is_returned(self, session):
+        session.app.post_json('/api/repos/ceph/', params=base_repo_data())
+        result = session.app.get(
+            '/api/search/',
+            params={'project': 'ceph', 'sha1': 'latest'},
+        )
+        assert result.json[0]['sha1'] == "45107e21c568dd033c2f0a3107dec8f0b0e58374"
+
+    def test_one_repo_does_not_match(self, session):
```

Comment:
Why do we need to assume the same ref? The API will use whatever filters the client specifies. If it specifies a ref, then it will use that, otherwise it will assume "a repo across any ref"